src/OrchardCore/OrchardCore.Abstractions/Shell/Builders/Extensions/ServiceProviderExtensions.cs

```diff
@@ -20,5 +22,19 @@ public static class ServiceProviderExtensions
         {
             return (TResult)ActivatorUtilities.CreateInstance(provider, type);
         }
+
+        /// <summary>
+        /// Gets the service object of the specified type with the specified key.
+        /// </summary>
+        public static object? GetKeyedService(this IServiceProvider provider, Type serviceType, object? serviceKey)
```

Comment:
@jtkech I'm not sure why we need this?!! we could the use the built-in extension method `GetKeyedService<T>()`

Reply:
Yes the generic `GetKeyedService<T>()` exists but the non generic is missing and is useful when the type is kwown at runtime, like we have `GetService<T>()` and `GetService()`.

There is a `ServiceProvider.GetKeyedService()` method but no `IServiceProvider` extension.

I was using this `GetKeyedService()` extension but for other reasons related to nullable context I finally used the existing `GetRequiredKeyedService()`, but still useful to have `GetKeyedService()`.

Reply:
yes that is useful method. Is it time to merge this one?

Reply:
I will change the namespace then I will merge

Reply:
> There is a ServiceProvider.GetKeyedService() method but no IServiceProvider extension.

It's in `IKeyedServiceProvider`

Reply:
@jtkech I changed the namespace to `Microsoft.Extensions.DependencyInjection.Extensions` to make it easier for discovery, if all are agree with this change I will merge

Reply:
I added the new extension in an existing class, so not good to change the namespace, it may break some code as you saw yourself by having to fix the build

Reply:
Ok, I will revert the latest changes then merge